src/logcollector/lccom.c

```diff
@@ -33,13 +34,32 @@ size_t lccom_dispatch(char * command, char ** output){
         }
         return lccom_getconfig(rcv_args, output);
 
+    } else if (strcmp(rcv_comm, "getstate") == 0) {
+        return lccom_getstate(output);
     } else {
         mdebug1("LCCOM Unrecognized command '%s'.", rcv_comm);
         os_strdup("err Unrecognized command", *output);
         return strlen(*output);
     }
 }
 
+size_t lccom_getstate(char ** output) {
+    cJSON * state_json = NULL;
```

Comment:
I would assign **_w_logcollector_state_get_**'s output here and would add the modifier "**const**" if this variable won't be modified